kong/plugins/statsd/log.lua

```diff
@@ -456,31 +454,12 @@ function _M.execute(conf)
   local message = kong.log.serialize({ngx = ngx, kong = kong, })
   message.cache_metrics = ngx.ctx.cache_metrics
 
-  local queue_id = kong.plugin.get_id()
-  local q = queues[queue_id]
-  if not q then
-    local batch_max_size = conf.queue_size or 1
-    local process = function (entries)
-      return log(conf, entries)
-    end
-
-    local opts = {
-      retry_count    = conf.retry_count or 10,
-      flush_timeout  = conf.flush_timeout or 2,
-      batch_max_size = batch_max_size,
-      process_delay  = 0,
-    }
-
-    local err
-    q, err = BatchQueue.new("statsd", process, opts)
-    if not q then
-      kong.log.err("could not create queue: ", err)
-      return
-    end
-    queues[queue_id] = q
-  end
-
-  q:add(message)
+  Queue.enqueue(
+    Queue.get_params(conf),
+    send_entries_to_upstream_server,
+    conf,
+    message
+  )
```

Comment:
About this, I am not sure about the order of parameters and requirement of passing the handler_conf, it seems only be passed forward.

```lua
function _M.execute(conf)
  Queue.enqueue(message, Queue.get_params(conf), function()
    -- I already have access to conf
  end)
end
```

You could also do:
```
function _M.execute(conf)
  Queue.enqueue(message, Queue.get_params(conf), send_entries_to_upstream_server, "cat", "dog")
end
```

which would mean it makes more sense to the callback this way:
```
local function send_entries_to_upstream_server(messages, arg1, arg1, ...)
```